app/admin/models/Categories_model.php

```diff
@@ -105,4 +106,80 @@ public function scopeIsEnabled($query)
     {
         return $query->where('status', 1);
     }
+    
+    public function scopeWhereHasLocation($query, $locationId)
+    {
+        $query->whereHas('locations', function ($q) use ($locationId) {
+            $q->where('locations.location_id', $locationId);
+        });
+    }
+	
+	 public function scopeListFrontEnd($query, $options = [])
+    {
+        extract(array_merge([
+            'page' => 1,
+            'pageLimit' => 20,
+            'enabled' => TRUE,
+            'sort' => 'id asc',
+            'group' => null,
+            'location' => null,            
+            'search' => '',
+            'orderType' => null,
+        ], $options));
+
+        $searchableFields = ['name', 'description'];
+
+        if (strlen($location) AND is_numeric($location)) {
+            $query->whereHasOrDoesntHaveLocation($location);
+            $query->with(['locations' => function ($q) use ($location) {
+                $q->whereHasOrDoesntHaveLocation($location);
+                $q->isEnabled();
+            }]);
+        }
+
+        if (strlen($location)) {
+            $query->whereHas('locations', function ($q) use ($location) {
+                $q->whereSlug($location);
+            });
+        }
```

Comment:
Not needed

Reply:
If I don't add lines of code, how do we filter the categories by Location slug?

Reply:
Ok, we want to filter by `whereHasOrDoesntHaveLocation`  or `whereSlug`, as is both will be applied to the query so maybe combine both if statements to an if/else and apply `whereHasOrDoesntHaveLocation` when location is an ID and `whereSlug` when location is a slug?

Reply:
finally left the filter for slug.